beacon-chain/rpc/eth/beacon/sync_committee.go

```diff
@@ -53,21 +54,21 @@ func (bs *Server) ListSyncCommittees(ctx context.Context, req *ethpbv2.StateSync
 				err,
 			)
 		}
+		if reqPeriodStartEpoch > currentPeriodStartEpoch+params.BeaconConfig().EpochsPerSyncCommitteePeriod {
+			return nil, status.Errorf(
+				codes.Internal,
+				"Could not fetch sync committee too far in the future. Requested epoch: %d, current epoch: %d",
+				*req.Epoch, currentEpoch,
+			)
+		}
 	}
 
-	if reqPeriodStartEpoch > currentPeriodStartEpoch+params.BeaconConfig().EpochsPerSyncCommitteePeriod {
-		return nil, status.Errorf(
-			codes.Internal,
-			"Could not fetch sync committee too far in the future. Epoch: %d",
-			*req.Epoch,
-		)
-	}
-
-	if reqPeriodStartEpoch > currentPeriodStartEpoch {
-		*req.Epoch = slots.ToEpoch(currentSlot)
+	stateEpoch := reqPeriodStartEpoch
+	if stateEpoch > currentPeriodStartEpoch+params.BeaconConfig().EpochsPerSyncCommitteePeriod {
+		stateEpoch = currentEpoch
 	}
 	st, err := bs.stateFromRequest(ctx, &stateRequest{
-		epoch:   req.Epoch,
+		epoch:   &stateEpoch,
```

Comment:
This change introduces a bug: if the user requests with `req.Epoch == nil` and with `req.stateId` a slot number, this method will return the current SyncCommittee instead of the requested one.

Reply:
Thanks. I misunderstood how `stateFromRequest` works